src/components/SignerPanel/SignerPanelContent.js

```diff
@@ -62,28 +65,45 @@ class ActionPathsContent extends React.Component {
       </span>
     )
   }
-  render() {
-    const { intent: { description, to, tx }, paths } = this.props
-    const { selected } = this.state
-    const radioItems = paths.map(({ appName, description }) => {
+  getPathRadioItem(path) {
+    // Slice off the intention (last transaction in the path)
+    path = path.slice(0, path.length - 1)
+
+    const title = path.map(({ name }, index) => {
       const shortName =
-        appName.length > RADIO_ITEM_TITLE_LENGTH
-          ? appName.slice(0, RADIO_ITEM_TITLE_LENGTH) + '…'
-          : appName
-      return {
-        description,
-        title: <span title={appName}>{shortName}</span>,
-      }
+        name.length > RADIO_ITEM_TITLE_LENGTH
+          ? name.slice(0, RADIO_ITEM_TITLE_LENGTH) + '…'
+          : name
+
+      const isLastSegment = index === path.length - 1
+      return <span title={name}>{shortName}{!isLastSegment ? ' → ' : ''}</span>
     })
-    const showPaths = !tx && paths.length
+
+    const isOneHop = path.length === 1
+    const description = path.map(({ name, description }, index) => {
+      if (isOneHop) return description
+
+      return `${index + 1}. ${name}: ${description}`
+    }).join('\n')
```

Comment:
Might have to do some funny `<br />` business instead of the `.join('\n')`.

Reply:
Nope, works when I tested it 🥇

Reply:
Hmm, just tried it and it's not showing up as new lines (only spaces) :(.

Reply:
Maybe I was just super lucky and the text wrapped because of an address or something :/